bpf/lib/nat.h

```diff
@@ -589,6 +589,8 @@ static __always_inline bool snat_v4_needed(struct __ctx_buff *ctx,
                        .saddr = ip4->saddr
                };
 
+	       *from_endpoint = true;
```

Comment:
This feels like a risky change. In old code we only set `from_endpoint = true` after checking `!is_reply`, but now it's unconditional.


I believe we should limit this change to the EgressGW path, as follows:

```
target->addr = egress_gw_policy->egress_ip;
target->egress_gateway = true;
if (local_ep)
	*from_endpoint = true;
```

If there's further opportunity for fine-tuning, let's review and push that separately.

Reply:
Ah nevermind - just realized that there's no other path in `snat_v4_needed()` that would return true. So even though we set the `from_endpoint` flag a lot earlier, the caller will only look at it under the same conditions as before. Think this is fine then :rocket:.